### PR TITLE
refactor: use lit observers

### PR DESCRIPTION
**Pull request by jeripeierSBB**

### CODE REVIEW

File: src/elements/flip-card/flip-card-details/flip-card-details.ts [hunk 1]
```diff
@@ -32,25 +40,6 @@ export class SbbFlipCardDetailsElement extends LitElement {
       .forEach((el: Element) => el.setAttribute(cardFocusableAttributeName, ''));
   }
 
-  public override connectedCallback(): void {
-    super.connectedCallback();
-    this._checkForSlottedActions();
```
Comment (kyubisation) [inline src/elements/flip-card/flip-card-details/flip-card-details.ts:37]: Is the call to _checkForSlottedActions no longer necessary?

Comment (jeripeierSBB) [reply src/elements/flip-card/flip-card-details/flip-card-details.ts:37]: yes, it's not needed as the observer does also check whenever its connected

File: src/elements/map-container/map-container.ts [hunk 2]
```diff
@@ -33,11 +31,36 @@ export class SbbMapContainerElement extends LitElement {
 
   @state() private _scrollUpButtonVisible = false;
 
-  private _intersector?: HTMLSpanElement;
   private _language = new SbbLanguageController(this);
-  private _observer = new AgnosticIntersectionObserver((entries) =>
-    this._toggleButtonVisibilityOnIntersect(entries),
-  );
+  private _observer = new IntersectionController(this, {
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/elements/map-container/map-container.ts:35]: This implementation was partially broken so I refactored. But yes, did a lot of testing ;)

File: src/elements/flip-card/flip-card-details/flip-card-details.ts [hunk 3]
```diff
@@ -32,25 +40,6 @@ export class SbbFlipCardDetailsElement extends LitElement {
...
-    this._flipCardMutationObserver.observe(this, {
-      childList: true,
-      subtree: true,
-    });
-  }
-
-  protected override firstUpdated(changedProperties: PropertyValues): void {
-    super.firstUpdated(changedProperties);
-    this._checkForSlottedActions();
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/elements/flip-card/flip-card-details/flip-card-details.ts:46]: Not needed as the observer lives from construction on and detects everything :)

File: src/elements-experimental/package.json [hunk 4]
```diff
@@ -14,6 +14,7 @@
     "@sbb-esta/lyne-elements": "0.0.0-PLACEHOLDER"
   },
   "dependencies": {
+    "@lit-labs/observers": "0.0.0-LITOBSERVERS",
```
Comment (kyubisation) [inline src/elements-experimental/package.json:17]: Do we have a usage here?

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Thank you!

**Pull request merged**